Source/ASDisplayNode.mm

```diff
@@ -447,14 +447,20 @@ - (void)asyncTraitCollectionDidChangeWithPreviousTraitCollection:(ASPrimitiveTra
       ASCornerRoundingType cornerRoundingType = _cornerRoundingType;
       UIColor *backgroundColor = _backgroundColor;
       if (_loaded(self)) {
+        CGColorRef cgBackgroundColor = backgroundColor.CGColor;
         if (self.isLayerBacked) {
           // Background colors do not dynamically update for layer backed nodes since they utilize CGColorRef
           // instead of UIColor. We utilize the _backgroundColor instance variable to track the full dynamic color
           // and apply any changes here when trait collection updates occur.
-          CGColorRef cgBackgroundColor = backgroundColor.CGColor;
           if (!CGColorEqualToColor(_layer.backgroundColor, cgBackgroundColor)) {
             _layer.backgroundColor = cgBackgroundColor;
           }
+        } else {
+          // we set view and layer background color in _ASPendingState, so do the same here
+          if (!CGColorEqualToColor(_layer.backgroundColor, cgBackgroundColor)) {
```

Comment:
Is it possible to move the color equality test up, so that the code doesn't have to be here twice?